src/Chart.jsx

```diff
@@ -351,13 +343,41 @@ class Chart extends Component {
       .attr('fill', 'none')
       .attr('pointer-events', 'all')
       .on('mouseover', () => {
+        g.selectAll('.mouse-per-line').attr('transform', function(d, i) {
+          var svg_elem = this.parentElement.parentElement.parentElement;
+          var foreign = document.createElementNS(
```

Comment:
DOM operations like `createElement`, `appendChild` could be done with d3 itself. May be even, the template could be moved to React, and rendered on hover.